src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,211 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> lastNode;
+    private Node<T> firstNode;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(lastNode, value, null);
+        if (lastNode == null) {
+            firstNode = node;
+            lastNode = node;
+            size++;
+        } else {
+            lastNode.next = node;
+            lastNode = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if ((index >= 0) && (index == size)) {
```

Comment:
```suggestion
        if (index == size) {
```
index >= 0 is already validated in checkIndexForAdd method